tests/VectorException/enc/enc.c

```diff
@@ -285,27 +285,29 @@ int enc_test_vector_exception()
     return 0;
 }
 
-uint64_t test_divide_by_zero_handler_with_ocall(
-    oe_exception_record_t* exception_record)
+void call_invalid_instruction()
 {
-    if (exception_record->code != OE_EXCEPTION_DIVIDE_BY_ZERO)
+    asm volatile("ud2;");
+}
+
+uint64_t test_sigill_handler_with_ocall(oe_exception_record_t* exception_record)
+{
+    if (exception_record->code != OE_EXCEPTION_ILLEGAL_INSTRUCTION)
```

Comment:
One good way to lock down the behavior would be to take a snapshot of the registers before the ocall and then after return from the ocall. The non-volatile registers ought to be preserved.

Reply:
Added to the test and modified idiv to only generate a 2 byte instruction

Reply:
Thanks.